src/api/app/controllers/webui/webui_controller.rb

```diff
@@ -342,4 +343,9 @@ def set_breadcrumbs
       { name: name, path: root_path }
     ]
   end
+
+  def set_pending_announcement
```

Comment:
@hennevogel I had changed the code before reading your last comment related to User#pending_announcement. Now, that method is not in the User model anymore, but in this WebuiController. Please check if it is better now. It simplifies the views too.

Please, take into account that all kind of users can see the ToS except the user that has already accepted it. `Nobody user` sees the ToS as well but not the button to accept it.

Reply:
Cool